### PR TITLE
Add network/codec tests

### PR DESCRIPTION
**Pull request by sshumakov**
Added unit tests for network/codec. #6

### CODE REVIEW

**Comment by awishformore** [comment]
I'm not sure it makes too much sense to test this, as the codec is likely to be expanded to include many more entities in the future.

**Comment by awishformore** [comment]
If you could change it to use the testify package, that would be cool.

**Comment by sshumakov** [comment]
Agree, testing all kind of messages that will be supported in the future can be too much. Lets keep those few tests for different message types only.

**Comment by sshumakov** [comment]
Integrated testify in all places, looks more readable indeed. Please take a look.

**Comment by sshumakov** [comment]
Good point, thanks for noticing.

File: network/codec_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    msg := 1
+    err := codec.Encode(buf, msg)
+
+    assert.NotNil(t, err, "Expected to receive error in case message type is unknown")
+}
+
+func TestDecodeUnknownCode(t *testing.T) {
+    codec := SimpleCodec{}
+
+    buf := make([]byte, 1)
```
Comment (awishformore) [inline network/codec_test.go]: I would not call `buf`, which is usually `bytes.Buffer` the same as the byte slice.
Usually, I would use `data` or `bytes` to make the distinction.

Comment (awishformore) [inline network/codec_test.go]: Also, in this case, you can just write `buf := []byte{77}` instead.

File: network/codec_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    _, ok := decoded.(*Discover)
+
+    assert.True(t, ok, "Decoded type is other than expected")
+}
+
+func TestEncodePeers(t *testing.T) {
+    codec := SimpleCodec{}
+
+    buf := bytes.NewBuffer(make([]byte, 0))
+    addrs := make([]string, 2)
```
Comment (awishformore) [inline network/codec_test.go]: `addrs := []string{"127.0.0.1", "192.168.4.62"}`

File: network/codec_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    assert.True(t, ok, "Decoded type is other than expected")
+    assert.Equal(t, msg, decodedMsg)
+}
+
+func TestEncodeBytes(t *testing.T) {
+    codec := SimpleCodec{}
+
+    buf := &bytes.Buffer{}   
+    msg := make([]byte, 2)
```
Comment (awishformore) [inline network/codec_test.go]: `msg := []byte{1,55}`

**Pull request merged**